packages/node/src/localEvaluation.ts

```diff
@@ -0,0 +1,184 @@
+    flags?: string[],
+  ): Promise<Variants> {
+    // Evaluate the flag configs and user.
+    const flagConfigs = await this.getFlagConfigs(flags);
+    this.debug('[Experiment] evaluate - user:', user, 'flags:', flags);
+    const resultsString = evaluation.evaluate(
+      JSON.stringify(flagConfigs),
+      JSON.stringify(user),
+    );
+    this.debug('[Experiment] evaluate - result:', resultsString);
+    // Parse variant results
+    const results: EvaluationResult = JSON.parse(resultsString);
+    const variants: Variants = {};
+    Object.keys(results).forEach((key) => {
+      variants[key] = {
+        value: results[key].variant.key,
+        payload: results[key].variant.payload,
+      };
+    });
+    return variants;
+  }
+
+  /**
+   * Start polling for flag configurations.
+   *
+   * You must call this function to begin polling for flag config updates.
+   * The promise returned by this function is resolved when the initial call
+   * to fetch the flag configuration completes.
+   *
+   * Calling this function while the poller is already running will stop and
+   * restart the poller.
+   */
+  public async start(): Promise<void> {
+    this.stop();
+    this.debug('[Experiment] poller - start');
+    this.flagConfigPoller = setInterval(async () => {
+      await this.updateFlagConfigs();
+    }, this.config.flagConfigPollingInterval);
+    if (!this.flagConfigPromise) {
+      this.flagConfigPromise = this.updateFlagConfigs({
+        attempts: 5,
+        min: 1,
+        max: 1,
+        scalar: 1,
+      });
+      await this.flagConfigPromise;
+    }
+  }
+
+  /**
+   * Stop polling for flag configurations.
+   *
+   * Calling this function while the poller is not running will do nothing.
+   */
+  public stop(): void {
+    if (this.flagConfigPoller) {
+      this.debug('[Experiment] poller - stop');
+      clearTimeout(this.flagConfigPoller);
+      this.flagConfigPoller = undefined;
+    }
+  }
+
+  private async fetchFlagConfigs(): Promise<Record<string, string>> {
+    const flagConfigs = await this.doFlagConfigs();
+    return this.parseFlagConfigs(flagConfigs);
+  }
+
+  private async getFlagConfigs(flagKeys?: string[]): Promise<string[]> {
+    if (this.flagConfigPromise) {
+      this.debug('[Experiment] waiting for flag configs');
+      await this.flagConfigPromise;
+    }
+    return Object.values(this.flagConfigCache.get(flagKeys));
+  }
+
+  private async updateFlagConfigs(
+    backoffPolicy?: BackoffPolicy,
+  ): Promise<void> {
+    this.debug('[Experiment] updating flag configs');
+    return await doWithBackoff<void>(async () => {
+      const flagConfigs = await this.fetchFlagConfigs();
+      this.flagConfigCache.clear();
+      this.flagConfigCache.put(flagConfigs);
+    }, backoffPolicy);
+  }
+
+  private async doFlagConfigs(): Promise<string> {
+    const endpoint = `${this.config.serverUrl}/sdk/rules?d=fdsa`;
+    const headers = {
+      Authorization: `Api-Key ${this.apiKey}`,
```

Comment:
also, thinking about types of Api-Key. This might need some discussion for JS-client in the future. Intention was that /rules endpoint will only respond to server/secret keys and not to client keys. Works fine for this case because both clients (local and remote) use the server key. But for client stuff in the future, we will have to think a little

Reply:
I dont think real client-side (web, mobile) code will call this function. this is a server-only sdk, and from our roadmap we dont have any plans to support client-side local evaluation at this point.

The only "client-side" local eval would be run on the edge which is technically still server-side.